functions/notify_slack.py

```diff
@@ -91,19 +95,20 @@ def lambda_handler(event, context):
             logger.info("S3 replication failed")
             notify_slack({"text": "Replication failed"})
         elif record["eventName"] == "ObjectCreated:Put":
-            guardduty_event = get_guardduty_event(
+            guardduty_events_list = get_guardduty_events(
                 record["s3"]["bucket"]["name"], record["s3"]["object"]["key"]
             )
             logger.info("GuardDuty event:")
-            logger.info(guardduty_event)
-            if "sample" in guardduty_event["service"]["additionalInfo"] and guardduty_event["service"]["additionalInfo"]["sample"] == True:
-                logger.info("SAMPLE EVENT")
-                guardduty_event["title"] = "[SAMPLE EVENT]" + guardduty_event["title"]
-                if "IGNORE_SAMPLE_EVENTS" in os.environ and os.environ["IGNORE_SAMPLE_EVENTS"] == "true":
-                    logger.info("SAMPLE EVENT SKIPPED")
-                    continue 
-            alert_payload = make_guardduty_alert_payload(guardduty_event)
-            result = notify_slack(alert_payload)
-            logger.info("HTTP Result:")
-            logger.info(result)
+            logger.info(guardduty_events_list)
```

Comment:
Would it make more sense to log each event as it's processed rather than in one-blob. That way we'd know if a particular event caused an error? e.g.

```
for event in list:
  logger.info(f'Event: {event}')
```

I'm also not sure if we *should* be logging sensitive events in the lambda logs. I guess it's OK? only ACP have access to them? Thoughts?

As an aside, not sure if this is a Python 2 / 3 environment. In which case, may not support the suggested f-string interpolation in the above example.

Reply:
Thanks, totally agree on the printing, I actually did that locally but forgot to commit.  I have also tidied up the other logging lines too.

I think it is a really good point on the sensitive data, I don't think it is extremely sensitive data and I believe ACP are the only ones with access to it, but there isn't any advantage on printing the whole json object, so I will only print out the ID.